src/components/MainArea.tsx

```diff
@@ -1,18 +1,55 @@
-import { useState } from "react";
+import { useEffect, useLayoutEffect, useState } from "react";
 import { ShadowDom } from "./ShadowDom.tsx";
 import CreateRecipe from "./recipes/CreateRecipe.tsx";
 import Recipes from "./recipes/Recipes.tsx";
-import { Route, Routes } from "react-router-dom";
+import { Route, Routes, useLocation, useNavigate } from "react-router-dom";
 import CheckRecipes from "./cart/CheckRecipes.tsx";
 
 const MainArea = () => {
   const [parentElement] = useState(() =>
-    document.querySelector("#pageFullWidth"),
+    document.querySelector<HTMLElement>("#pageFullWidth")
   );
 
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  useEffect(() => {
+    let oldHref: null | string = null;
+    const body = document.querySelector("body");
+
+    const observer = new MutationObserver(() => {
+      const { pathname } = new URL(document.location.href);
+
+      if (pathname !== oldHref || oldHref === null) {
+        oldHref = pathname;
+        if (pathname !== "/") {
+          navigate("/");
+          console.log("naviguji na defaultni route, kde nic nerenderujeme...");
+        }
+      }
+    });
+
+    if (body) {
+      observer.observe(body, { childList: true, subtree: true });
+    }
+
+    return () => observer.disconnect();
+  }, [navigate]);
+
+  useLayoutEffect(() => {
+    // Google Analytics
+    if (!parentElement) {
+      return;
+    }
+
+    parentElement.style.display = location.pathname === "/" ? "block" : "none";
+  }, [location, parentElement]);
```

Comment:
Tohle na zaklade naseho routeru (kdykoli kdy je nejaka navigace, zmeni se location) nastavuje style domu, kde to rohlik renderuje, nemuzem jim to zmazat, jinak jim z toho blazni javascript :D